sql/dd/impl/bootstrap/bootstrapper.cc

```diff
@@ -1211,6 +1213,17 @@ bool initialize_dd_properties(THD *thd) {
       LogErr(ERROR_LEVEL, ER_LCTN_CHANGED, lower_case_table_names, actual_lctn);
       return true;
     }
+
+    // read actual dd engine from dd_properties
+    uint actual_dd_engine = DB_TYPE_INNODB;
```

Comment:
Maybe initialize to something != INNODB, != ROCKSDB to catch corrupted reads?

Reply:
add another check for actual_dd_engine value for corrupted read.. 

another way is that initialize to unknown first, then if the property doesn't exist, initialize to innodb?

Reply:
> add another check for actual_dd_engine value for corrupted read..
> 
> another way is that initialize to unknown first, then if the property doesn't exist, initialize to innodb?

I'd prefer to fail early on such corruption instead of trying to overcome it silently, but my operations background is weak, hence no strong opinion

Reply:
> another way is that initialize to unknown first, then if the property doesn't exist, initialize to innodb?

+1